src/nccl_ofi_net.c

```diff
@@ -908,8 +959,11 @@ static int nccl_net_ofi_domain_release(nccl_net_ofi_domain_t *domain)
 
 	nccl_net_ofi_mutex_lock(&domain->domain_lock);
 
-	if (HASH_COUNT(domain->endpoint_table) == 0) {
-		nccl_net_ofi_mutex_lock(&device->device_lock);
+	if (HASH_COUNT(domain->endpoint_table) == 0 || skip_device_lock) {
```

Comment:
Why does setting skip_device_lock mean we want to force cleanup?  That's counter-intuitive to me.